sm/download/src/download.rs

```diff
@@ -95,12 +111,40 @@ impl Downloader {
     }
 }
 
+fn pre_allocate_space(file: &File, file_len: u64) -> Result<(), DownloadError> {
+    let tmpstats = statvfs::statvfs("/tmp")?;
```

Comment:
The path provided in this call should not be hardcoded, this could likely ease testing as well.
The downloader uses path provided in the new method which is `download_target`, so maybe you also want to path directory path for which you will do statvfs.

Reply:
Indeed, this must be derived from the file path.

A better type signature might be:

```
fn pre_allocate_space(path: &Path, file_len: u64) -> Result<File, DownloadError>
```

Reply:
fixed